projects/microphysics/scripts/score_training.py

```diff
@@ -71,42 +110,45 @@ def main(config: TrainConfig, seed: int = 0, model_url: str = None):
     test_ds = config.open_dataset(
         config.train_url, config.nfiles, config.model_variables
     )
+    n = 80_000
+    train_set = next(iter(train_ds.unbatch().shuffle(2 * n).batch(n)))
+    test_set = next(iter(test_ds.unbatch().shuffle(2 * n).batch(n)))
 
-    train_set = next(iter(train_ds.unbatch().shuffle(160_000).batch(80_000)))
-    test_set = next(iter(test_ds.unbatch().shuffle(160_000).batch(80_000)))
+    summary_metrics = {}
+    profiles = {}
 
-    train_scores, train_profiles = score_model(model, train_set)
-    test_scores, test_profiles = score_model(model, test_set)
-    logger.debug("Scoring Complete")
+    for split_name, data in [("train", train_set), ("test", test_set)]:
+        scores, profiles = score_model(model, data)
+        summary_metrics.update(
+            {f"score/{split_name}/{key}": value for key, value in scores.items()}
+        )
+        class_scores, class_profiles = score_gscond_classes(config, model, data)
+        scores.update(class_scores)
 
-    summary_metrics: Dict[str, Any] = {
-        f"score/train/{key}": value for key, value in train_scores.items()
-    }
-    summary_metrics.update(
-        {f"score/test/{key}": value for key, value in test_scores.items()}
-    )
+        for score, value in scores.items():
+            summary_metrics[f"score/{split_name}/{score}"] = value
+
+        all_profiles = {**profiles, **class_profiles}
+        # add level for dataframe index, assumes equivalent feature dims
+        sample_profile = next(iter(all_profiles.values()))
+        all_profiles["level"] = np.arange(len(sample_profile))
+
+        if config.use_wandb:
+            log_to_table(f"profiles/{split_name}", all_profiles)
 
+    logger.debug("Scoring Complete")
+
+    # log scalar metrics
     # Logging for google cloud
     log_fact_json(data=summary_metrics)
+    if config.use_wandb:
+        config.wandb.job.log(summary_metrics)
 
+    # log individual profile predictions
     if config.use_wandb:
         pred_sample = model.predict(test_set, batch_size=8192)
         log_profile_plots(test_set, pred_sample)
 
-        # add level for dataframe index, assumes equivalent feature dims
-        sample_profile = next(iter(train_profiles.values()))
-        train_profiles["level"] = np.arange(len(sample_profile))
-        test_profiles["level"] = np.arange(len(sample_profile))
-
-        config.wandb.job.log(summary_metrics)
-
-        log_to_table("profiles/train", train_profiles)
-        log_to_table("profiles/test", test_profiles)
-
-    with put_dir(config.out_url) as tmpdir:
```

Comment:
i deleted this since the data is stored in wandb and I  wanted to de-duplicate the logic for train and test.